psc-package/Main.hs

```diff
@@ -0,0 +1,259 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+
+module Main where
+
+import qualified Data.Aeson as Aeson
+import           Data.Aeson.Encode.Pretty
+import           Data.Foldable (fold, for_, traverse_)
+import           Data.List (nub)
+import qualified Data.Map as Map
+import           Data.Maybe (mapMaybe)
+import qualified Data.Set as Set
+import           Data.Text (pack)
+import qualified Data.Text.Lazy as TL
+import qualified Data.Text.Lazy.Builder as TB
+import           Data.Text.Encoding (encodeUtf8)
+import           Data.Traversable (for)
+import           Data.Version (showVersion)
+import qualified Filesystem.Path.CurrentOS as Path
+import           GHC.Generics (Generic)
+import qualified Options.Applicative as Opts
+import qualified Paths_purescript as Paths
+import qualified System.IO as IO
+import           Turtle hiding (fold)
+
+packageFile :: Path.FilePath
+packageFile = "psc-package.json"
+
+data PackageConfig = PackageConfig
+  { name    :: Text
+  , depends :: [Text]
+  , set     :: Text
+  , source  :: Text
+  } deriving (Show, Generic, Aeson.FromJSON, Aeson.ToJSON)
+
+pathToTextUnsafe :: Turtle.FilePath -> Text
+pathToTextUnsafe = either (error "Path.toText failed") id . Path.toText
+
+defaultPackage :: Text -> PackageConfig
+defaultPackage pkgName =
+  PackageConfig { name    = pkgName
+                , depends = [ "prelude" ]
+                , set     = "psc-" <> pack (showVersion Paths.version)
```

Comment:
As far as I can tell, this is the only part of `psc-package` that currently uses the `purescript` library. Distributing `psc-package` like this as a part of the compiler suite does seem like a good way of getting package sets to match up with the installed compiler without needing people to write it in, and has the additional advantage that it wouldn't make the installation instructions any longer, but have we considered whether it would make sense to develop `psc-package` as a separate tool i.e. outside this repository?